### PR TITLE
Reimplement dashboards with terraform-plugin-framework

### PR DESCRIPTION
**Pull request by tosuke**
Output from acceptance testing:

```
$ MACKEREL_EXPERIMENTAL_TFFRAMEWORK=1 make testacc TESTS='"Acc(DataSource)?MackerelDashboard"'
TF_ACC=1 go test -v ./mackerel/... -run "Acc(DataSource)?MackerelDashboard" -timeout 120m                                                                                                          2024/09/06 12:50:53 [INFO] mackerel: use terraform-plugin-framework based implementation
=== RUN   TestAccDataSourceMackerelDashboardGraph
=== PAUSE TestAccDataSourceMackerelDashboardGraph
=== RUN   TestAccDataSourceMackerelDashboardValue
=== PAUSE TestAccDataSourceMackerelDashboardValue
=== RUN   TestAccDataSourceMackerelDashboardMarkdown
=== PAUSE TestAccDataSourceMackerelDashboardMarkdown
=== RUN   TestAccDataSourceMackerelDashboardAlertStatus
=== PAUSE TestAccDataSourceMackerelDashboardAlertStatus
=== RUN   TestAccMackerelDashboardGraph
=== PAUSE TestAccMackerelDashboardGraph
=== RUN   TestAccMackerelDashboardValue
=== PAUSE TestAccMackerelDashboardValue
=== RUN   TestAccMackerelDashboardMarkdown
=== PAUSE TestAccMackerelDashboardMarkdown
=== RUN   TestAccMackerelDashboardAlertStatus
=== PAUSE TestAccMackerelDashboardAlertStatus
=== CONT  TestAccDataSourceMackerelDashboardGraph
=== CONT  TestAccMackerelDashboardMarkdown
=== CONT  TestAccDataSourceMackerelDashboardAlertStatus
=== CONT  TestAccDataSourceMackerelDashboardMarkdown
=== CONT  TestAccMackerelDashboardAlertStatus
=== CONT  TestAccMackerelDashboardValue
=== CONT  TestAccMackerelDashboardGraph
=== CONT  TestAccDataSourceMackerelDashboardValue
--- PASS: TestAccDataSourceMackerelDashboardMarkdown (13.84s)
--- PASS: TestAccDataSourceMackerelDashboardAlertStatus (13.85s)
--- PASS: TestAccDataSourceMackerelDashboardGraph (14.78s)
--- PASS: TestAccDataSourceMackerelDashboardValue (18.29s)
--- PASS: TestAccMackerelDashboardMarkdown (19.34s)
--- PASS: TestAccMackerelDashboardValue (19.96s)
--- PASS: TestAccMackerelDashboardAlertStatus (20.02s)
--- PASS: TestAccMackerelDashboardGraph (20.50s)
PASS
ok      github.com/mackerelio-labs/terraform-provider-mackerel/mackerel 21.649s
```

### CODE REVIEW

File: internal/provider/resource_mackerel_dashboard.go [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+				Computed:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"graph":        schemaDashboardResource_graph,
+			"value":        schemaDashboardResource_value,
+			"markdown":     schemaDashboardResource_markdown,
+			"alert_status": schemaDashboardResource_alertStatus,
+		},
+	}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_dashboard.go:178]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/2918d8df36ce55b21a5ca4b3bcc3ee115b19d796/mackerel/resource_mackerel_dashboard.go?plain=1#L143-L342

File: internal/provider/data_source_mackerel_dashboard.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+				Computed:    true,
+			},
+			"created_at": schema.Int64Attribute{
+				Description: schemaDashboardCreatedAtDesc,
+				Computed:    true,
+			},
+			"updated_at": schema.Int64Attribute{
+				Description: schemaDashboardUpdatedAtDesc,
+				Computed:    true,
+			},
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_dashboard.go:99]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/2918d8df36ce55b21a5ca4b3bcc3ee115b19d796/mackerel/data_source_mackerel_dashboard.go?plain=1#L137-L336

File: go.mod [hunk 3]
```diff
@@ -5,7 +5,7 @@ go 1.22.0
 require (
 	github.com/golangci/golangci-lint v1.50.1
 	github.com/google/go-cmp v0.6.0
-	github.com/hashicorp/terraform-plugin-framework v1.8.0
+	github.com/hashicorp/terraform-plugin-framework v1.11.0
```
Comment (tosuke) [inline go.mod:8]: Embedded struct in a schema is supported from v1.11.0 (FYI: https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v1.11.0).

File: internal/provider/resource_mackerel_dashboard.go [hunk 4]
```diff
@@ -0,0 +1,576 @@
...
+
+			"host": schema.ListNestedBlock{
+				Description: schemaDashboardGraph_hostDesc,
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.ExactlyOneOf(
+						path.MatchRelative(),
+						path.MatchRelative().AtParent().AtName("role"),
+						path.MatchRelative().AtParent().AtName("service"),
+						path.MatchRelative().AtParent().AtName("service"),
```
Comment (yohfee) [inline internal/provider/resource_mackerel_dashboard.go:315]: Is this an intentional duplication?

**Review by yohfee — COMMENTED** [review]
Since this is a migration, is it okay to leave the things that are not supported from the original as they are?
For example, there are `valueRange`, `formatRules`, `referenceLines` and so on in help page.

File: internal/provider/resource_mackerel_dashboard.go (hunk 4 above)
Comment (tosuke) [reply internal/provider/resource_mackerel_dashboard.go]: Not intentional. Fixed it: https://github.com/mackerelio-labs/terraform-provider-mackerel/pull/239/commits/15d4a2014c7f568ded7376ed8eff59af54c006b8

**Comment by tosuke** [comment]
@yohfee These features are not yet supported because they are outside the scope of this PR. When importing a dashboard that uses these features, some of the settings will be lost, so an error will be displayed to inform the user(https://github.com/mackerelio-labs/terraform-provider-mackerel/pull/239/commits/2145cb5d943340111cae2be72e856bac304f7783).

**Review by yohfee — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**